src/Serilog.Sinks.RollingFile/Sinks/RollingFile/TemplatedPathRoller.cs

```diff
@@ -97,26 +154,68 @@ public IEnumerable<RollingLogFile> SelectMatches(IEnumerable<string> filenames)
                 if (match.Success)
                 {
                     var inc = 0;
-                    var incGroup = match.Groups["inc"];
+                    var incGroup = match.Groups[MatcherMarkInc];
                     if (incGroup.Captures.Count != 0)
                     {
                         var incPart = incGroup.Captures[0].Value.Substring(1);
                         inc = int.Parse(incPart, CultureInfo.InvariantCulture);
                     }
 
-                    DateTime date;
-                    var datePart = match.Groups["date"].Captures[0].Value;
+                    DateTime dateTime;
+                    var dateTimePart = match.Groups[MatcherMarkSpecifier].Captures[0].Value;
                     if (!DateTime.TryParseExact(
-                        datePart,
-                        DateFormat,
+                        dateTimePart,
+                        _usedFormat,
                         CultureInfo.InvariantCulture,
                         DateTimeStyles.None,
-                        out date))
+                        out dateTime))
                         continue;
 
-                    yield return new RollingLogFile(filename, date, inc);
+                    yield return new RollingLogFile(filename, dateTime, inc);
                 }
             }
         }
+
+        public DateTime GetCurrentCheckpoint(DateTime date)
+        {
+            if (_specifierType == SpecifierTypeEnum.Hour)
+            {
+                return date.Date.AddHours(date.Hour);
+            }
+            else if (_specifierType == SpecifierTypeEnum.HalfHour)
+            {
+                DateTime auxDT = date.Date.AddHours(date.Hour);
+                if (date.Minute >= 30)
+                    auxDT = auxDT.AddMinutes(30);
+                return auxDT;
+            }
+
+            return date.Date;
+        }
+
+        public DateTime GetNextCheckpoint(DateTime date)
+        {
+            DateTime currentCheckpoint = GetCurrentCheckpoint(date);
+
+            if (_specifierType == SpecifierTypeEnum.Hour)
+            {
+                return currentCheckpoint.AddHours(1);
+            }
+            else if (_specifierType == SpecifierTypeEnum.HalfHour)
+            {
+                return currentCheckpoint.AddMinutes(30);
+            }
+
+            return currentCheckpoint.AddDays(1);
+        }
     }
-} 
\ No newline at end of file
+
+    enum SpecifierTypeEnum
```

Comment:
If we decide to keep the enum (see comment above), we should drop the `Enum` suffix. Just calling it `Specifier` would be clear enough.